onnxscript/converter.py

```diff
@@ -1231,6 +1231,8 @@ def translate_loop_stmt(self, loop_stmt: Union[ast.For, ast.While]):
         )
         for pv in loop_state_vars:
             ov = self.py_var_to_onnx_var(pv, debuginfo.DebugInfo(loop_stmt, self))
+            if ov not in self.current_fn.assigned_names:
```

Comment:
Possible to add a comment on the situation this is handling? I think that would help future readers a lot

Reply:
Good point. Added some comments.